lib/features/artifacts.js

```diff
@@ -128,7 +128,22 @@ export default class Artifacts {
       };
 
       try {
-        await uploadArtifact(taskHandler, stream, entryName, expiry, headers);
+        await new Promise((accept, reject) => {
+          let numRetries = taskHandler.runtime.retry.numRetries;
+          let retry = () => {
+            uploadArtifact(taskHandler, stream, entryName, expiry, headers)
```

Comment:
This does the `createArtifact` call, it's unlikely that there is a need to retry that (signed url doesn't expire in 5 min)... but I guess okay to do...

It's probably easiest to do all of this inside `uploadArtifact` and use a temporary file.